src/js/helpers.js

```diff
@@ -516,33 +498,19 @@ ${enterpriseSettings}`;
     const jwtToken = getToken();
     installScriptArgs = `${installScriptArgs} --commercial --jwt-token "${jwtToken}"`;
   }
+  let menderSetupArgs = `--quiet --device-type "${deviceType}"`
+  if (isHosted) {
+    menderSetupArgs = `${menderSetupArgs} --demo --hosted-mender --tenant-token "${tenantToken}"`;
+  } else if (isEnterprise) {
+    menderSetupArgs = `${menderSetupArgs} --retry-poll 30 --update-poll 5 --inventory-poll 5 --server-url https://${window.location.hostname} --server-cert="" --tenant-token "${tenantToken}"`;
```

Comment:
Right. So there is some implementations details I had to check with `mender setup`.

In short, `--demo` can be used with `--hosted`, in which case it means "setup the client for demo polling intervals but using a production server", or with `--server-url`, in which case it means "setup the client for demo polling intervals and demo server, modifying `/etc/hosts` and installing the server demo certificate".

Hence I corrected the original code to use `--demo` for HM (this is better than specifying the actual intervals, as it will let `mender` decide on what are the optimal settings (for example adding also long timeouts for Update Control Maps), and specifying the intervals for Enterprise (unfortunately Update Control Maps timeouts are not set for this case).

Another simplification that I have done is assuming that Enterprise will always have a CA trusted certificate. If it is not the case the user can re-run `mender setup` to specify the path. Without this (aka as it was before) the call won't be unattended and the user will be prompted for the cert path.

Reply:
Thanks for the explanation and for checking the details 👌!